tests/cli.rs

```diff
@@ -144,3 +144,19 @@ fn run_it<S: AsRef<str>>(
 
     Ok(())
 }
+
+#[rstest(
+number, expected_output,
+case(0, ""),
+case(2, "{foo: bar, abc: [123, 456]}\n{foo: baz, abc: [420d-1, 4.3e1]}"),
+)]
+///Calls ion-cli beta head with different requested number. Pass the test if the return value equals to the expected value.
+fn test_head(number: i32, expected_output: &str) -> Result<()> {
```

Comment:
Prefer the [modern `rstest` syntax](https://docs.rs/rstest/latest/rstest/#creating-parametrized-tests).

```suggestion
#[rstest]
#[case(0, "")]
#[case(2, "{foo: bar, abc: [123, 456]}\n{foo: baz, abc: [420d-1, 4.3e1]}")]
///Calls ion-cli beta head with different requested number. Pass the test if the return value equals to the expected value.
fn test_head(#[case] number: i32, #[case] expected_output: &str) -> Result<()> {
```

Reply:
Separately, I'd rather see this as a test for `write_all_values`, or whatever we call it now :)